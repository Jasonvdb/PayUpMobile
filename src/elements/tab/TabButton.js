import { StyleSheet, TouchableOpacity } from "react-native";
import Icon from "react-native-vector-icons/SimpleLineIcons";
import React from "react";
import PropTypes from "prop-types";

import theme from "../../config/theme";

const TabButton = ({
	route,
	onPress,
	onLongPress,
	getAccessibilityLabel,
	isActive,
	iconString
}) => (
	<TouchableOpacity
		style={styles.tabButton}
		onPress={() => {
			onPress(route);
		}}
		onLongPress={() => {
			onLongPress(route);
		}}
		accessibilityLabel={getAccessibilityLabel({
			route
		})}
	>
		<Icon
			name={iconString}
			size={25}
			color={isActive ? theme.brand1 : theme.disabled1}
		/>
	</TouchableOpacity>
);

TabButton.propTypes = {
	route: PropTypes.object.isRequired,
	onPress: PropTypes.func.isRequired,
	onLongPress: PropTypes.func.isRequired,
	getAccessibilityLabel: PropTypes.func.isRequired,
	isActive: PropTypes.bool.isRequired,
	iconString: PropTypes.string.isRequired
};

export default TabButton;

const styles = StyleSheet.create({
	tabButton: {
		flex: 1,
		display: "flex",
		justifyContent: "center",
		height: "100%",
		alignItems: "center"
	}
});
