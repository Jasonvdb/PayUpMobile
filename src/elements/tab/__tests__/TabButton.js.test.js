/**
 * @format
 */

import "react-native";
import React from "react";
import { TouchableOpacity } from "react-native";
import { shallow, render } from "enzyme";

import TabButton from "../TabButton";

const testProps = {
	route: {},
	onPress: () => {},
	onLongPress: () => {},
	getAccessibilityLabel: () => {},
	isActive: true,
	iconString: "user"
};

describe("TabButton", () => {
	const origConsole = console.error;
	beforeEach(() => {
		console.error = message => {
			if (message.indexOf("PascalCase" > -1)) {
				return;
			}

			origConsole(message);
		};
	});
	afterEach(() => {
		console.error = origConsole;
	});
	
	describe("Rendering", () => {
		it("should match to snapshot ", () => {
			const inactiveComponent = render(
				<TabButton {...testProps} isActive={false}/>,
			);
			expect(inactiveComponent).toMatchSnapshot("Inactive TabButton");

			const activeComponent = render(<TabButton {...testProps} isActive/>);
			expect(activeComponent).toMatchSnapshot("Active TabButton");
		});
	});

	describe("Interaction", () => {
		describe("onPress", () => {
			it("should call onPress", () => {
				const mockOnPress = jest.fn();
				const wrapper = shallow(
					<TabButton {...testProps} onPress={mockOnPress}/>,
				);
				wrapper
					.find(TouchableOpacity)
					.first()
					.props()
					.onPress();

				expect(mockOnPress).toHaveBeenCalled();
				expect(mockOnPress).toHaveBeenCalledTimes(1);
			});
		});

		describe("onLongPress", () => {
			it("should call onLongPress", () => {
				const mockOnPress = jest.fn();
				const wrapper = shallow(
					<TabButton {...testProps} onLongPress={mockOnPress}/>,
				);
				wrapper
					.find(TouchableOpacity)
					.first()
					.props()
					.onLongPress();

				expect(mockOnPress).toHaveBeenCalled();
				expect(mockOnPress).toHaveBeenCalledTimes(1);
			});
		});
	});
});
