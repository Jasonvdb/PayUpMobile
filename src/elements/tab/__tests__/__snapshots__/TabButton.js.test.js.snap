// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton Rendering should match to snapshot : Active TabButton 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "flex:1;display:flex;justify-content:center;height:100%;align-items:center;opacity:1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "0:[object Object];2:[object Object];3:[object Object]",
        },
        "children": Array [
          Object {
            "data": "",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "text",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`TabButton Rendering should match to snapshot : Inactive TabButton 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "flex:1;display:flex;justify-content:center;height:100%;align-items:center;opacity:1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "0:[object Object];2:[object Object];3:[object Object]",
        },
        "children": Array [
          Object {
            "data": "",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "text",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
